code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm

```diff
@@ -0,0 +1,175 @@
+
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/find_valid_home,
+		/datum/ai_planning_subtree/enter_exit_home,
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree,
+		/datum/ai_planning_subtree/find_and_hunt_target/pollinate,
+	)
+
+/datum/ai_planning_subtree/find_valid_home
+
+/datum/ai_planning_subtree/find_valid_home/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	var/mob/living/work_bee = controller.pawn
+
+	var/obj/structure/beebox/current_home = controller.blackboard[BB_CURRENT_HOME]
+
+	if(QDELETED(current_home))
+		controller.queue_behavior(/datum/ai_behavior/find_and_set/bee_hive, BB_CURRENT_HOME, /obj/structure/beebox)
+		return
+
+	if(work_bee in current_home.bees)
+		return
+
+	controller.queue_behavior(/datum/ai_behavior/inhabit_hive, BB_CURRENT_HOME)
+	return SUBTREE_RETURN_FINISH_PLANNING
+
+/datum/ai_planning_subtree/enter_exit_home
+	///chance we go back home
+	var/flyback_chance = 15
+	///chance we exit the home
+	var/exit_chance = 35
+
+/datum/ai_planning_subtree/enter_exit_home/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+
+	var/obj/structure/beebox/current_home = controller.blackboard[BB_CURRENT_HOME]
+
+	if(QDELETED(current_home))
+		return
+
+	var/mob/living/bee_pawn = controller.pawn
+	var/action_prob =  (bee_pawn in current_home.contents) ? exit_chance : flyback_chance
+
+	if(!SPT_PROB(action_prob, seconds_per_tick))
+		return
+
+	controller.queue_behavior(/datum/ai_behavior/enter_exit_hive, BB_CURRENT_HOME)
+	return SUBTREE_RETURN_FINISH_PLANNING
+
+
+/datum/ai_planning_subtree/find_and_hunt_target/pollinate
+	target_key = BB_TARGET_HYDRO
+	hunting_behavior = /datum/ai_behavior/hunt_target/unarmed_attack_target/bees
+	finding_behavior = /datum/ai_behavior/find_hunt_target/pollinate
+	hunt_targets = list(/obj/machinery/hydroponics)
+	hunt_range = 7
+	//hunt_chance = 10
+
+
+/datum/ai_behavior/hunt_target/unarmed_attack_target/bees
+	always_reset_target = TRUE
+
+/datum/ai_behavior/find_hunt_target/pollinate
+
+/datum/ai_behavior/find_hunt_target/pollinate/valid_dinner(mob/living/source, obj/machinery/hydroponics/dinner, radius)
+	if(!dinner.myseed)
+		return FALSE
+	if(dinner.plant_status == HYDROTRAY_PLANT_DEAD || dinner.recent_bee_visit)
+		return FALSE
+
+	return can_see(source, dinner, radius)
+
+/datum/ai_behavior/enter_exit_hive
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_REQUIRE_REACH
+
+/datum/ai_behavior/enter_exit_hive/setup(datum/ai_controller/controller, target_key)
+	. = ..()
+	var/atom/target = controller.blackboard[target_key]
+	if(QDELETED(target))
+		return FALSE
+	set_movement_target(controller, target)
+
+/datum/ai_behavior/enter_exit_hive/perform(seconds_per_tick, datum/ai_controller/controller, target_key)
+	. = ..()
+	var/obj/structure/beebox/current_home = controller.blackboard[target_key]
+	var/mob/living/bee_pawn = controller.pawn
+
+	bee_pawn.UnarmedAttack(current_home)
```

Comment:
Unarmed attack? I think we should split that behavior out of UnarmedAttack out into a new board and then store it as a `/datum/callback` in the blackboard or something that we can execute since we definitely know that this is going to execute the enter hive behavior. Going to have really odd consequences if this behavior is drag and dropped anywhere else. Would allow it to be infinitely more extensible to more mobs that might need to pop in and out of hives without being reliant on unarmed attack.

Reply:
i putted him in the unarmedatack so if the bee is sentient he can interacted with the objects the ai can interacted with

Reply:
Forgot to elaborate, this would be done in conjunction with the other suggestion that the earlier behavior should be split out into a signal (or at least invoked via `melee_attack()`) so that both the player bee and the AI bee can use it by either of those interactions.